cmd/snap/cmd_set.go

```diff
@@ -116,6 +117,11 @@ func (x *cmdSet) Execute([]string) error {
 				// Not valid JSON-- just save the string as-is.
 				patchValues[parts[0]] = parts[1]
 			} else {
+				err := validateJSONKeys(value)
+				if err != nil {
+					return fmt.Errorf("invalid configuration key: %w", err)
```

Comment:
I would just return the error to avoid saying the same thing twice:

```go
if err := validateJSONKeys(value); err != nil {
   return err
}
```

Reply:
Done! Thank you!